workflow.py

```diff
@@ -0,0 +1,27 @@
+import utils
+from search.search_engine import SearchEngine
+import constants
+
+def query_workflow(search, query):
+    short_gresponse = utils.from_gcloud_remove_extra(utils.gcloud_syntax_extraction(query))
+
+    keywords = []
+
+    for token in short_gresponse['tokens']:
+        keywords.append(token['lemma'])
+
+    lan = short_gresponse['lan']
+    synonyms = []
+    if lan == 'es':
+        for word in keywords:
+            for synonym in utils.create_synonym_list_esp(word):
+                synonyms.append(synonym)
+
+    return search.query(keywords, synonyms)
+
+def main():
+    search = SearchEngine(constants.ARTICLES, keywords_weight=2)
+    print(query_workflow(search, constants.QUERY_ARTICLE_1))
```

Comment:
This is a bit awkward, because it forces the end-user to figure out their own query_workflow to extract keywords & synonyms, but really the SearchEngine class can handle that all for you.

I would  flip around your `search.query` and `query_workflow` functions, so that your API is simply `search.query("this is a raw string query")`, and the first thing `search.query` does is invoke a helper method like SearchEngline class helper method like `query_workflow` that extracts keywords and synonyms.